### DISCUSSION TITLE
App: Aggregate functions on filter

### DISCUSSION THREAD

**Discussion by joselcvarela** [directus/directus#11239] [Feature Requests]
## Propose
Have aggregate functions on App side

## Motivation
Instead of filtering records per value, it would be useful to filter by count, average, min, max, etc. of values.

## Implications
This would need to have a new UI for filter interface.
On API side I don't think any change will be needed, since this is already used on Insights

Not sure if this agrees with [80/20 rule](https://docs.directus.io/contributing/introduction/#feature-requests) because it seems no one asked for it before. But just wanted to note it 👍

**Comment by jamescammarano** [comment]
Why would this need a new UI interface? Couldn't it just be more options in the drop-down?

**Reply by joselcvarela** [reply]
I think it's not possible because right now we have: 
```
Field - Operation - Value
```

So if you want to filter results by the AVG of a field greater than a value, you would at least need an extra option on filters operator.
For example, list results where `avg` of `foo` is `greater than` `100`.
```
Field: foo
Operation: greater than
Value: 100
```
As you can see, we can't specify `avg`.

Although we could implements something like `avg(foo)` on Field. 🤔

**Comment by rijkvanzanten** [comment]
The interesting thing about this is that you don't use "average" on the value of a single row. You average out a group of rows. So you can't really "find items where the average of X is Y", but instead you'd end up with "Group items by A, and only show the groups where the average of X is Y" 🤔

**Reply by joselcvarela** [reply]
Yeah, you're absolutely right. 
If I remember correctly, the intent of this idea was to filter based on amount of relationships, for example, filter categories with 5 or more products, and I think this behaviour was added here https://github.com/directus/directus/pull/12488 so huge thank you for that ❤️

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
